packages/bridge-ui/src/components/modals/FaucetModal.svelte

```diff
@@ -1,41 +1,35 @@
 <script lang="ts">
-  import { BigNumber, ethers, logger, Signer } from "ethers";
-  import { TEST_ERC20 } from "../../domain/token";
-  import { pendingTransactions } from "../../store/transactions";
-  import { signer } from "../../store/signer";
-  import { errorToast, successToast } from "../../utils/toast";
-  import { _ } from "svelte-i18n";
-  import MintableERC20 from "../../constants/abi/MintableERC20";
-  import { fromChain } from "../../store/chain";
-  import { fetchSigner, switchNetwork } from "@wagmi/core";
-  import { CHAIN_MAINNET } from "../../domain/chain";
-  import Modal from "./Modal.svelte";
-  import { onMount } from "svelte";
-  import { token } from "../../store/token";
+  import { BigNumber, ethers, Signer } from 'ethers';
+  import { pendingTransactions } from '../../store/transactions';
+  import { signer } from '../../store/signer';
+  import { errorToast, successToast } from '../../utils/toast';
+  import { _ } from 'svelte-i18n';
+  import MintableERC20 from '../../constants/abi/MintableERC20';
+  import { fromChain } from '../../store/chain';
+  import { fetchSigner, switchNetwork } from '@wagmi/core';
+  import { CHAIN_MAINNET } from '../../domain/chain';
+  import Modal from './Modal.svelte';
+  import { onMount } from 'svelte';
+  import { token } from '../../store/token';
 
   export let isOpen: boolean = false;
   export let onMint: () => Promise<void>;
 
   let disabled: boolean = true;
 
-  onMount(async () => {
-    isBtnDisabled($signer);
-  });
-
-  $: isBtnDisabled($signer).catch((e) => console.error(e));
-
   async function isBtnDisabled(signer: Signer) {
     if (!signer) return;
+
     const balance = await signer.getBalance();
+    const address = await signer.getAddress();
+
     const contract = new ethers.Contract(
-      TEST_ERC20.addresses[0].address,
+      $token.addresses[0].address,
```

Comment:
is $token guaranteed to be set?

Reply:
I renamed the function `isBtnDisabled()` since it was a bit misleading (does not return a boolean) to `shouldEnableButton` which set the internal state `disabled`, and handled the possibility of not having a token selected